src/app/Http/Controllers/Admin/LaravelLogController.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use App\Http\Controllers\Controller;
+use Symfony\Component\HttpFoundation\BinaryFileResponse;
+use Illuminate\Http\Request;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Support\Facades\Gate;
+
+class LaravelLogController extends Controller
+{
+    public function laravelLog(Request $request): JsonResponse|BinaryFileResponse
+    {
+        $user = $request->user();
+        if (!$user->isAdmin()) {
+            return response()->json(['error' => 'Unauthorized.'], 403);
+        }
+
+        $filePath = storage_path('logs/laravel.log');
+
+        if (file_exists($filePath)) {
+            return response()->download($filePath);
```

Comment:
Does laravel have a good way to gzip the outgoing bytes?

Reply:
didnt see anything magical with like arrows and shit